errorreporting/errorreporting_quickstart/main.go

```diff
@@ -32,30 +33,34 @@ var errorClient *errorreporting.Client
 func main() {
 	ctx := context.Background()
 
-	// Sets your Google Cloud Platform project ID.
-	projectID := "YOUR_PROJECT_ID"
+	// TODO: Sets your Google Cloud Platform project ID via environment or explicitly
+	projectID := os.Getenv("GOOGLE_PROJECT_ID")
```

Comment:
Why add this? Also, if you are going to add it, `GOOGLE_CLOUD_PROJECT` is the standard environment variable, not `GOOGLE_PROJECT_ID`.

Reply:
We try to align samples across _all_ languages to exhibit the same behavior. The options are:

* Environment variable
* Explicit setup
* Fallback to project ID implicitly configured at client initialization

Unfortunately I did not have time to dig for implementation of the last bullet in Golang. I will appreciate it if you can point me to the code that "retrieves" project ID from the client settings.

Reply:
+1 to changing to `GOOGLE_CLOUD_PROJECT`
```suggestion
	projectID := os.Getenv("GOOGLE_CLOUD_PROJECT")
```
... primarily because the rest of this repository uses `GOOGLE_CLOUD_PROJECT` ([see search results](https://github.com/search?q=repo%3AGoogleCloudPlatform%2Fgolang-samples+%22GOOGLE_CLOUD_PROJECT%22&type=code)).
There's no use of `GOOGLE_PROJECT_ID`.

Reply:
I am not sure autodetecting the project ID is supported in all clients for Go; see discussion on googleapis/google-cloud-go#1294 . @codyoss please correct me if I'm wrong.

In snippet samples we generally pass the project ID in via an arg (see https://googlecloudplatform.github.io/samples-style-guide/#minimal-arguments ). As this is a quickstart, I think it's fine to try to detect it from the environment. However I do think we should standardize on `GOOGLE_CLOUD_PROJECT` rather than `GOOGLE_PROJECT_ID` given that is what is supported across tools.

Reply:
Changes made to address this while allowing users to provide the project ID via argument.